src/main/java/org/openstreetmap/atlas/generator/AtlasGeneratorHelper.java

```diff
@@ -560,6 +561,45 @@ public String getName()
         };
     }
 
+    protected static PairFunction<Tuple2<String, Atlas>, String, Atlas> subatlas(
+            final ConfiguredFilter filter, final AtlasCutType cutType)
+    {
+        return tuple ->
+        {
+            final Atlas subAtlas;
+            // Grab the tuple contents
+            final String shardName = tuple._1();
+            final Atlas originalAtlas = tuple._2();
+            logger.info("Starting sub Atlas for for Atlas {}", originalAtlas.getName());
+            final Time start = Time.now();
+            try
+            {
+                // Slice the Atlas
+                final Optional<Atlas> subAtlasOptional = originalAtlas.subAtlas(filter::test,
+                        cutType);
+                if (subAtlasOptional.isPresent())
+                {
+                    subAtlas = subAtlasOptional.get();
+                }
+                else
+                {
+                    subAtlas = null;
```

Comment:
I would make this a `PairFlatMapFunction` so as to return nothing (an empty list) when this is null, and a one item list when this is not. Then there is no risk to have null values in the RDD.
-- Unless I am missing something and null elements are already handled downstream?

Reply:
Null elements are filtered on line 345 of AtlasGenerator!